x-pack/plugins/alerting/server/routes/create_rule.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { validateDurationSchema, ILicenseState, AlertTypeDisabledError } from '../lib';
+import { CreateOptions } from '../alerts_client';
+import {
+  RewriteRequestCase,
+  RewriteResponseCase,
+  handleDisabledApiKeysError,
+  verifyAccessAndContext,
+} from './lib';
+import {
+  SanitizedAlert,
+  validateNotifyWhenType,
+  AlertTypeParams,
+  AlertingRequestHandlerContext,
+  BASE_ALERTING_API_PATH,
+  AlertNotifyWhenType,
+} from '../types';
+
+export const bodySchema = schema.object({
+  name: schema.string(),
+  rule_type_id: schema.string(),
+  enabled: schema.boolean({ defaultValue: true }),
+  consumer: schema.string(),
+  tags: schema.arrayOf(schema.string(), { defaultValue: [] }),
+  throttle: schema.nullable(schema.string({ validate: validateDurationSchema })),
+  params: schema.recordOf(schema.string(), schema.any(), { defaultValue: {} }),
+  schedule: schema.object({
+    interval: schema.string({ validate: validateDurationSchema }),
+  }),
+  actions: schema.arrayOf(
+    schema.object({
+      group: schema.string(),
+      id: schema.string(),
+      params: schema.recordOf(schema.string(), schema.any(), { defaultValue: {} }),
```

Comment:
The legacy `create` body schema had `actionTypeId` inside the `actions` object. Should there be a `connector_type_id` field now instead?

Reply:
Ah, yes. This field is no longer necessary as part of the create rule API. There is code to denormalize this id in the alerts client while it validates access/existence https://github.com/elastic/kibana/blob/master/x-pack/plugins/alerting/server/alerts_client/alerts_client.ts#L1456.